app/services/valid_test_data_generators/applications_populater.rb

```diff
@@ -94,17 +110,48 @@ def reject_application(application)
       application.reload
     end
 
+    def defer_application(application)
+      Participants::Defer.new(lead_provider:,
+                              participant: application.user,
+                              course_identifier: application.course.identifier,
+                              reason: Participants::Defer::DEFERRAL_REASONS.sample).defer
+    end
+
+    def withdrawn_application(application)
+      Participants::Withdraw.new(lead_provider:,
+                                 participant: application.user,
+                                 course_identifier: application.course.identifier,
+                                 reason: Participants::Withdraw::WITHDRAWAL_REASONS.sample).withdraw
+    end
+
     def create_declarations(application)
       %w[started retained-1 retained-2 completed].each do |declaration_type|
         break if Faker::Boolean.boolean(true_ratio: 0.5)
 
         FactoryBot.create(
           :declaration,
-          :submitted_or_eligible,
+          state: Declaration.states.keys.sample,
           application:,
           declaration_type:,
         )
       end
     end
+
+    def create_outcomes(application)
```

Comment:
Is this used?

Reply:
👍  Rebased with main and now it's all good.. tks @ethax-ross